vendor/bat-native-ads/src/bat/ads/internal/ml/data/vector_data.cc

```diff
@@ -150,13 +206,47 @@ void VectorData::Normalize() {
   }
 }
 
-int VectorData::GetDimensionCountForTesting() const {
+int VectorData::GetDimensionCount() const {
   return storage_->dimension_count();
 }
 
+int VectorData::GetNonZeroElementsCount() {
+  if (!storage_->dimension_count()) {
+    return 0;
+  }
+
+  int non_zero_count = 0;
+  size_t v_index = 0;
+  while (v_index < storage_->GetSize()) {
+    if (storage_->values()[v_index] != 0) {
+      non_zero_count += 1;
+    }
+    ++v_index;
+  }
+  return non_zero_count;
+}
+
 const std::vector<float>& VectorData::GetValuesForTesting() const {
   return storage_->values();
 }
 
+const std::string VectorData::GetVectorAsString() const {
+  if (!storage_->dimension_count()) {
+    return "";
+  }
+
+  std::string vctr;
```

Comment:
nit: rename to `vector_as_string`. Maybe also initialise it in L234 and return in line 235 for copy elision

Reply:
Renamed